### DISCUSSION TITLE
OS migration from Windows 2012 to 2019

### DISCUSSION THREAD

**Discussion by roggss** [nextgenhealthcare/connect#5866] [Q&A]
Hi,

Anyone can tell me about the best approach/process in moving Mirth to another server?

Below is what I have in mind.

1. Stop Mirth on the old server.
2. Back up the config file in Mirth from old server
3. Install the same Mirth version to new server.
4. Restore the Mirth config file on the new server.
5. Start Mirth on the new server.

I am using Oracle as DB., so don't need to copy folder from Mirth old server. right?
Let me know if something I missed or should have done.

Thanks
Rey

**Comment by JoeFox82** [comment]
That's the way I do it, too. Don't forget the .properties files from the conf & appdata folder and the vmoptions files from the root path of the installation :-)

**Reply by roggss** [reply]
Hi,

The entire app data folder or jus the .properties file on the appdata folder?

I see that there is a server.id on that appdata folder,, is that needed to be moved?

**Reply by pacmano1** [reply]
If you don't persist the server.id and your using the same DB, your stats get lost.   Also, if you license plugins you will need to address that with Nextgen.

**Reply by roggss** [reply]
okay so everything on the appdata folder then.

I dont have license plugins. 

Thanks

**Reply by jonbartels** [reply]
Don't forget the `conf` directory!

**Reply by wheaton3000** [reply]
I'm trying to do a migration at this moment.  How do you do step 4, restore the Mirth config on the new server?

**Reply by roggss** [reply]
@wheaton3000 if you click on "Settings" on the "Server Task" panel, you should see the menu "Restore Config"

**Reply by wheaton3000** [reply]
@roggss 

I haven't successfully completed the migration yet, but I think these are the correct steps.  What do you think?

1. Backup the config file on the old Mirth server by going into 'Settings', 'Server Task' panel and selecting 'Backup Config'
2. Stop Mirth on the old server
3. Install Mirth on new server
4. Log into the Mirth Administrator on the new server and click on 'Settings', 'Server Task' panel and select 'Restore Config'
5. Shut down the Mirth service on the new server
6. Copy the 'appdata', 'conf' folders and the vmoption files to the correct directories from the old server to the new server
7. Start up the Mirth service on the new service
